src/main/kotlin/org/onflow/flow/sdk/crypto/Crypto.kt

```diff
@@ -156,11 +159,53 @@ object Crypto {
 }
 
 internal class HasherImpl(
-    private val hashAlgo: HashAlgorithm
+    private val hashAlgo: HashAlgorithm,
+    private val key: ByteArray? = null,
+    private val customizer: ByteArray? = null,
+    private val outputSize: Int? = null
 ) : Hasher {
+    private var kmac: KMAC? = null
+
+    init {
+        if (hashAlgo == HashAlgorithm.KMAC128) {
+            if (key == null || key.size < 16) {
+                throw IllegalArgumentException("KMAC128 requires a key of at least 16 bytes")
+            }
+            if (outputSize != null && outputSize <= 0) {
+                throw IllegalArgumentException("Output size must be positive")
+            }
+            kmac = KMAC(128, customizer)
+            kmac!!.init(KeyParameter(key))
+        }
+    }
+
     override fun hash(bytes: ByteArray): ByteArray {
-        val digest = MessageDigest.getInstance(hashAlgo.algorithm)
-        return digest.digest(bytes)
+        return when (hashAlgo) {
+            HashAlgorithm.KECCAK256 -> {
+                val keccakDigest = Keccak.Digest256()
+                keccakDigest.digest(bytes)
+            }
+            HashAlgorithm.KMAC128 -> {
+                val output = ByteArray(outputSize ?: kmac!!.digestSize)
+                kmac!!.update(bytes, 0, bytes.size)
+                kmac!!.doFinal(output, 0)
```

Comment:
I am not sure how this function behaves when `output` is a buffer larger than `32` (which seems to be the default output size). I am concerned `doFinal` would only fill the first 32 bytes of `output` and keeps the rest empty.
My suspicion comes from the fact that there is a second `doFinal` [variant](https://downloads.bouncycastle.org/java/docs/bcprov-jdk18on-javadoc/) with 3 parameters that explicitly takes the output length as a parameter. 

```
kmac!!.doFinal(output, 0, outputSize)
```
The sanity test uses 32 bytes as output size so it wouldn't catch the issue if there is one.

Maybe we should use the 3 params variant to be safe?